test/selenium/sessions_ui_test.js

```diff
@@ -47,115 +54,69 @@ describe('sessions ui', function() {
     });
 
     describe('signup', function() {
-      var elements;
-      var screens;
 
-      beforeEach(function() {
-        screens = {
-          signup: driver.findElement(webdriver.By.id('signup')),
-          signupSuccess: driver.findElement(webdriver.By.id('signup-success'))
-        };
-
-        elements = {
-          pwd1: driver.findElement(webdriver.By.id('signup-pwd1')),
-          pwd2: driver.findElement(webdriver.By.id('signup-pwd2')),
-          set: driver.findElement(webdriver.By.id('signup-button'))
-        };
-      });
+        beforeEach(function() {
+            setUpWebapp = new SetUpWebapp(driver);
+            setUpPage = setUpWebapp.getSetUpView();
+        });
 
       it('should show the signup screen', function() {
-        return driver.wait(webdriver.until.elementIsVisible(screens.signup),
-                           3000);
-      });
-
-      it('should not show the signupSuccess screen', function(done) {
-        screens.signupSuccess.isDisplayed().then(function(visible) {
-          assert.equal(visible, false);
-          done();
+          return setUpPage.isSetUpView();
         });
-      });
 
-      it('should have the right fields', function () {
-        var types = {
-          pwd1: 'password',
-          pwd2: 'password',
-          set: 'submit'
-        };
-        var promises = Object.keys(elements).map(function(key) {
-          return elements[key].getAttribute('type').then(function(value) {
```

Comment:
:warning: This function doesn't check if the password field hides passwords anymore. As the modification might be heavy, I'd recommend to leave this test as is for now and change it in a follow up.